src/main/services/connectors.service.ts

```
@@ -647,17 +678,17 @@ export default class ConnectorsService {
-  ): Promise<boolean | BigQueryTestResponse> {
+  ): Promise<ConnectorTestResponse> {
     await this.validateConnection(connection);
     switch (connection.type) {
       case 'postgres':
         return testPostgresConnection(connection);
       case 'snowflake':
-        try {
-          return await testSnowflakeConnection(connection);
-        } catch {
-          return false;
+        const result = await testSnowflakeConnection(connection);
```

Biome (2.5.3)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)